--- a/var/www/rdcore/cake4/rd_cake/src/Application.php
+++ b/var/www/rdcore/cake4/rd_cake/src/Application.php
@@ -80,7 +80,6 @@ class Application extends BaseApplication implements AuthenticationServiceProvid
          * Debug Kit should not be installed on a production system
          */
         if (Configure::read('debug')) {
-            $this->addPlugin('DebugKit');
         }
 
         // Load more plugins here
