--- a/var/www/rdcore/cake4/rd_cake/config/MESHdesk.php
+++ b/var/www/rdcore/cake4/rd_cake/config/MESHdesk.php
@@ -106,8 +106,8 @@ $config['common_node_settings']['report_adv_full']	        = 600;
 $config['common_node_settings']['report_adv_sampling']	    = 60;
  
 //Default MQTT Settings
-$config['mqtt_settings']['enable_realtime']	= false;
-$config['mqtt_settings']['api_gateway_url']	= 'http://204.48.30.220:8001';
+$config['mqtt_settings']['enable_realtime']	= env("RD_MQTT_SETTINGS_ENABLE_REALTIME", "false");
+$config['mqtt_settings']['api_gateway_url']	= env("RD_MQTT_SETTINGS_API_GATEWAY_URL", "http://127.0.0.1:8001");
 
 //== MESHdesk SSID/BSSID
 $config['MESHdesk']['bssid'] = "02:CA:FE:CA:00:00"; //This will be the first one; subsequent ones will be incremented 
--- a/var/www/rdcore/cake4/rd_cake/config/RadiusDesk.php
+++ b/var/www/rdcore/cake4/rd_cake/config/RadiusDesk.php
@@ -217,7 +217,7 @@ TIP: Also remember to change the values in:
 ApProfiles.php
 MESHdesk.php
 */
-$config['DynamicClients']['shared_secret'] = 'testing123';
+$config['DynamicClients']['shared_secret'] = env("RD_RADIUS_SECRET", "testing123");
 
 //=== EXPERIMENTAL STUFF =====
 //--Show experimental menus---
--- a/var/www/rdcore/cake4/rd_cake/config/app_local.php
+++ b/var/www/rdcore/cake4/rd_cake/config/app_local.php
@@ -40,13 +40,13 @@ return [
      */
     'Datasources' => [
         'default' => [
-            'host'          => 'localhost',
+            'host'          => env('RD_DB_SERVER','rdmariadb'),
             'className'     => 'Cake\Database\Connection',
             'driver'        => 'Cake\Database\Driver\Mysql',
             'persistent'    => false,
-            'username'      => 'rd',
-            'password'      => 'rd',
-            'database'      => 'rd',
+            'username'      => env('RD_DB_USERNAME','rd'),
+            'password'      => env('RD_DB_PASSWORD','rd'),
+            'database'      => env('RD_DB_NAME','rd'),
             // Comment out the line below if you are using PostgreSQL
             ////'encoding' => 'utf8mb4',
          	'encoding'      => 'utf8',
@@ -81,7 +81,7 @@ return [
         */
         
         'cake3' => [
-            'host' => 'localhost',
+            'host' => env('RD_DB_SERVER','rdmariadb'),
             'className' => 'Cake\Database\Connection',
             'driver' => 'Cake\Database\Driver\Mysql',
             'persistent' => false,
@@ -103,7 +103,7 @@ return [
          */
          /* //OCT 2022 disable this one for Docker image to work//
         'test' => [
-            'host' => 'localhost',
+            'host' => env('RD_DB_SERVER','rdmariadb'),
             //'port' => 'non_standard_port_number',
             'username' => 'my_app',
             'password' => 'secret',
@@ -123,7 +123,7 @@ return [
      */
     'EmailTransport' => [
         'default' => [
-            'host' => 'localhost',
+            'host' => env('RD_DB_SERVER','rdmariadb'),
             'port' => 25,
             'username' => null,
             'password' => null,
--- a/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/flows_wip.php
+++ b/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/flows_wip.php
@@ -1,8 +1,9 @@
 <?php
 //Some global variables
-$servername = "localhost";
-$username   = "rd";
-$password   = "rd";
+$servername = getenv("RD_DB_SERVER") ?: "rdmariadb";
+$username   = getenv("RD_DB_USERNAME") ?: "rd";
+$password   = getenv("RD_DB_PASSWORD") ?: "rd";
+$dbname     = getenv("RD_DB_NAME") ?: "rd";
 $conn       = false;
 $conn2      = false;
 
@@ -96,9 +97,9 @@ function main(){
   
 function doConnection(){
 
-    global $servername,$username,$password,$conn;
+    global $servername,$username,$password,$conn,$dbname;
     try {
-        $conn = new PDO("mysql:host=$servername;dbname=rd", $username, $password,[PDO::ATTR_PERSISTENT => true]);
+        $conn = new PDO("mysql:host=$servername;dbname=$dbname", $username, $password,[PDO::ATTR_PERSISTENT => true]);
         $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
     }
     catch(PDOException $e){
--- a/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/process_flows.php
+++ b/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/process_flows.php
@@ -2,9 +2,10 @@
 
 
 //Some global variables
-$servername = "localhost";
-$username   = "rd";
-$password   = "rd";
+$servername = getenv("RD_DB_SERVER") ?: "rdmariadb";
+$username   = getenv("RD_DB_USERNAME") ?: "rd";
+$password   = getenv("RD_DB_PASSWORD") ?: "rd";
+$dbname     = getenv("RD_DB_NAME") ?: "rd";
 $conn       = false;
 $conn2      = false;
 
@@ -74,9 +75,9 @@ function main(){
 }
 
 function doConnection(){
-    global $servername,$username,$password,$conn;
+    global $servername,$username,$password,$conn,$dbname;
     try {
-        $conn = new PDO("mysql:host=$servername;dbname=rd", $username, $password,[PDO::ATTR_PERSISTENT => true]);
+        $conn = new PDO("mysql:host=$servername;dbname=$dbname", $username, $password,[PDO::ATTR_PERSISTENT => true]);
         $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
     }
     catch(PDOException $e){
--- a/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/process_report.php
+++ b/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/process_report.php
@@ -6,9 +6,10 @@
 
 
 //Some global variables
-$servername = "localhost";
-$username   = "rd";
-$password   = "rd";
+$servername = getenv("RD_DB_SERVER") ?: "rdmariadb";
+$username   = getenv("RD_DB_USERNAME") ?: "rd";
+$password   = getenv("RD_DB_PASSWORD") ?: "rd";
+$dbname     = getenv("RD_DB_NAME") ?: "rd";
 $conn       = false;
 $conn2      = false;
 
@@ -84,7 +85,7 @@ function main(){
 //==== FOR Postgresql =====
 /*
 function doConnection(){
-    global $servername,$username,$password,$conn;
+    global $servername,$username,$password,$conn,$dbname;
     try {
         $conn = new PDO("pgsql:host=$servername;dbname=rd", $username, $password);
         $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
@@ -99,9 +100,9 @@ function doConnection(){
 //==== For Mysql / MariaDB =====
 function doConnection(){
 
-    global $servername,$username,$password,$conn;
+    global $servername,$username,$password,$conn,$dbname;
     try {
-        $conn = new PDO("mysql:host=$servername;dbname=rd", $username, $password,[PDO::ATTR_PERSISTENT => true]);
+        $conn = new PDO("mysql:host=$servername;dbname=$dbname", $username, $password,[PDO::ATTR_PERSISTENT => true]);
         $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);   
         $conn->setAttribute( PDO::ATTR_EMULATE_PREPARES, false );
     }
--- a/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/reporting.php
+++ b/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/reporting.php
@@ -1,8 +1,9 @@
 <?php
 //Some global variables
-$servername = "localhost";
-$username   = "rd";
-$password   = "rd";
+$servername = getenv("RD_DB_SERVER") ?: "rdmariadb";
+$username   = getenv("RD_DB_USERNAME") ?: "rd";
+$password   = getenv("RD_DB_PASSWORD") ?: "rd";
+$dbname     = getenv("RD_DB_NAME") ?: "rd";
 $conn       = false;
 $conn2      = false;
 
@@ -93,7 +94,7 @@ function main(){
 //==== FOR Postgresql =====
 /*
 function doConnection(){
-    global $servername,$username,$password,$conn;
+    global $servername,$username,$password,$conn,$dbname;
     try {
         $conn = new PDO("pgsql:host=$servername;dbname=rd", $username, $password);
         $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
@@ -107,9 +108,9 @@ function doConnection(){
 //==== For Mysql / MariaDB =====
 function doConnection(){
 
-    global $servername,$username,$password,$conn;
+    global $servername,$username,$password,$conn,$dbname;
     try {
-        $conn = new PDO("mysql:host=$servername;dbname=rd", $username, $password,[PDO::ATTR_PERSISTENT => true]);
+        $conn = new PDO("mysql:host=$servername;dbname=$dbname", $username, $password,[PDO::ATTR_PERSISTENT => true]);
         $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
     }
     catch(PDOException $e){
--- a/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/reporting_no_qmi.php
+++ b/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/reporting_no_qmi.php
@@ -1,8 +1,9 @@
 <?php
 //Some global variables
-$servername = "localhost";
-$username   = "rd";
-$password   = "rd";
+$servername = getenv("RD_DB_SERVER") ?: "rdmariadb";
+$username   = getenv("RD_DB_USERNAME") ?: "rd";
+$password   = getenv("RD_DB_PASSWORD") ?: "rd";
+$dbname     = getenv("RD_DB_NAME") ?: "rd";
 $conn       = false;
 $conn2      = false;
 
@@ -83,9 +84,9 @@ function main(){
   
 function doConnection(){
 
-    global $servername,$username,$password,$conn;
+    global $servername,$username,$password,$conn,$dbname;
     try {
-        $conn = new PDO("mysql:host=$servername;dbname=rd", $username, $password,[PDO::ATTR_PERSISTENT => true]);
+        $conn = new PDO("mysql:host=$servername;dbname=$dbname", $username, $password,[PDO::ATTR_PERSISTENT => true]);
         $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
     }
     catch(PDOException $e){
--- a/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/reporting_wip.php
+++ b/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/reporting_wip.php
@@ -1,8 +1,9 @@
 <?php
 //Some global variables
-$servername = "localhost";
-$username   = "rd";
-$password   = "rd";
+$servername = getenv("RD_DB_SERVER") ?: "rdmariadb";
+$username   = getenv("RD_DB_USERNAME") ?: "rd";
+$password   = getenv("RD_DB_PASSWORD") ?: "rd";
+$dbname     = getenv("RD_DB_NAME") ?: "rd";
 $conn       = false;
 $conn2      = false;
 
@@ -73,9 +74,9 @@ function main(){
   
 function doConnection(){
 
-    global $servername,$username,$password,$conn;
+    global $servername,$username,$password,$conn,$dbname;
     try {
-        $conn = new PDO("mysql:host=$servername;dbname=rd", $username, $password,[PDO::ATTR_PERSISTENT => true]);
+        $conn = new PDO("mysql:host=$servername;dbname=$dbname", $username, $password,[PDO::ATTR_PERSISTENT => true]);
         $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
     }
     catch(PDOException $e){
--- a/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/send_remote_syslog.php
+++ b/var/www/rdcore/cake4/rd_cake/setup/scripts/reporting/send_remote_syslog.php
@@ -2,9 +2,10 @@
 
 
 //Some global variables
-$servername = "localhost";
-$username   = "rd";
-$password   = "rd";
+$servername = getenv("RD_DB_SERVER") ?: "rdmariadb";
+$username   = getenv("RD_DB_USERNAME") ?: "rd";
+$password   = getenv("RD_DB_PASSWORD") ?: "rd";
+$dbname     = getenv("RD_DB_NAME") ?: "rd";
 $conn       = false;
 $conn2      = false;
 
@@ -77,9 +78,9 @@ function main(){
 
 function doConnection(){
 
-    global $servername,$username,$password,$conn;
+    global $servername,$username,$password,$conn,$dbname;
     try {
-        $conn = new PDO("mysql:host=$servername;dbname=rd", $username, $password,[PDO::ATTR_PERSISTENT => true]);
+        $conn = new PDO("mysql:host=$servername;dbname=$dbname", $username, $password,[PDO::ATTR_PERSISTENT => true]);
         $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
     }
     catch(PDOException $e){
