--- a/sites-available/radiusdesk-intra-id-default	2025-07-05 14:53:50.513853205 -0700
+++ b/sites-available/radiusdesk-intra-id-default	2025-07-05 14:54:46.840472985 -0700
@@ -581,6 +581,9 @@
     if((&request:WLAN-AKM-Suite)&&(&request:WLAN-AKM-Suite==1027074)){
         pl_ppsk          
     }
+    if((&request:WLAN-AKM-Suite)&&(&request:WLAN-AKM-Suite==1027076)){
+        pl_ppsk_ft
+    }
     RADIUSdesk_preacct
 
     #RADIUSdesk  We add this since some clients give a total wrong value thus messing the accounting detail
--- a/policy.d/radiusdesk	2025-07-05 14:59:00.002842569 -0700
+++ b/policy.d/radiusdesk	2025-07-05 14:59:37.164043613 -0700
@@ -5,6 +5,9 @@
 	if((&request:WLAN-AKM-Suite)&&(&request:WLAN-AKM-Suite==1027074)){
         pl_ppsk          
     }
+        if((&request:WLAN-AKM-Suite)&&(&request:WLAN-AKM-Suite==1027076)){
+        pl_ppsk_ft
+    }
 
     #We created a less strict "filter_username" since the default one from FreeRADIUS are very strict in terms of the Realm
     #Which it assumes is part of the username
