--- a/mods-config/perl/ppsk.pl	2025-04-11 04:54:09.000000000 -0700
+++ b/mods-config/perl/ppsk.pl	2025-08-07 20:44:34.612671542 -0700
@@ -89,10 +89,10 @@
 
 #___ RADIUSdesk _______
 sub read_conf {
-   $conf{'db_name'}     = "rd";
-   $conf{'db_host'}     = "127.0.0.1";
-   $conf{'db_user'}     = "rd";
-   $conf{'db_passwd'}   = "rd";
+   $conf{'db_name'}     = $ENV{RD_DB_NAME} || "rd";
+   $conf{'db_host'}     = $ENV{RD_DB_SERVER} || "rdmariadb";
+   $conf{'db_user'}     = $ENV{RD_DB_USERNAME} || "rd";
+   $conf{'db_passwd'}   = $ENV{RD_DB_PASSWORD} || "rd";
 }
 
 sub conn_db {
@@ -268,12 +268,9 @@
     my $FR_EAPoL_Key_Msg='';
     my $FR_Calling_Station='';
 
-    #if(($RAD_REQUEST{'Attr-245.26.11344.1'})&&($RAD_REQUEST{'Attr-245.26.11344.2'})){    
-    if(($RAD_REQUEST{'FreeRADIUS-802.1X-Anonce'})&&($RAD_REQUEST{'FreeRADIUS-802.1X-EAPoL-Key-Msg'})){    
-	#$FR_Anonce           = $RAD_REQUEST{'Attr-245.26.11344.1'};
-        $FR_Anonce           = $RAD_REQUEST{'FreeRADIUS-802.1X-Anonce'};
-	#$FR_EAPoL_Key_Msg    = $RAD_REQUEST{'Attr-245.26.11344.2'};
-        $FR_EAPoL_Key_Msg    = $RAD_REQUEST{'FreeRADIUS-802.1X-EAPoL-Key-Msg'};
+    if(($RAD_REQUEST{'Attr-245.26.11344.1'}||$RAD_REQUEST{'FreeRADIUS-802.1X-Anonce'})&&($RAD_REQUEST{'Attr-245.26.11344.2'}||$RAD_REQUEST{'FreeRADIUS-802.1X-EAPoL-Key-Msg'})){
+        $FR_Anonce           = $RAD_REQUEST{'Attr-245.26.11344.1'} || $RAD_REQUEST{'FreeRADIUS-802.1X-Anonce'};
+        $FR_EAPoL_Key_Msg    = $RAD_REQUEST{'Attr-245.26.11344.2'} || $RAD_REQUEST{'FreeRADIUS-802.1X-EAPoL-Key-Msg'};
 	    $FR_Calling_Station  = $RAD_REQUEST{'Calling-Station-Id'};
 		$FR_Anonce              =~ s/^0x//i;
 		$FR_EAPoL_Key_Msg       =~ s/^0x//i;   
@@ -307,42 +304,49 @@
 
     if(defined $RAD_REQUEST{'NAS-Identifier'} && length $RAD_REQUEST{'NAS-Identifier'} > 0) {
     
-        my $realm_id = 0;
+        my $realm_id_tmp = 0;
+        my @realm_ids = ();
+        if ( ! $dbh->ping ) {
+            CLONE();
+        }
         $stmt_realm_id->execute($RAD_REQUEST{'NAS-Identifier'});     
         while(my $row = $stmt_realm_id->fetchrow_hashref()){        
-            $realm_id = $row->{'id'};
+            $realm_id_tmp = $row->{'id'};
+            push(@realm_ids, $realm_id_tmp);
         }
         $stmt_realm_id->finish();
         
-        if(($ssid)&&($realm_id)){
-            &radiusd::radlog("2", "Found Realm ID $realm_id and ssid $ssid. We can try to get the realm_ssid ID");
-            
-            #Get the realm_ssid id
-            my $ssid_id = 0;
-            $stmt_ssid_id->execute($ssid,$realm_id);        
-            while(my $row = $stmt_ssid_id->fetchrow_hashref()){        
-                $ssid_id = $row->{'id'};
-            }
-            $stmt_ssid_id->finish();
-            
-            if($ssid_id){
-                &radiusd::radlog("2", "Found Realm ID $realm_id and ssid_id $ssid_id. We can try to get the LIST OF PPSKs");
-                $stmt_pmk_list->execute($ssid_id,$realm_id); 
-                my $match_found = 0;            
-                while(my $row = $stmt_pmk_list->fetchrow_hashref()){
-                    if(process_row($EAPOL1,$EAPOL2,$ssid,$row->{'ppsk'},$row)){
-                        #Formulate the reply
-                        formulate_reply($row,$realm_id);
-                        $match_found = 1;
-                        last;
-                    }
+        if(($ssid)&&($realm_id_tmp)){
+            my $match_found = 0;            
+            OUTERLOOP: for my $realm_id (@realm_ids) {
+                &radiusd::radlog("2", "Found Realm ID $realm_id and ssid $ssid. We can try to get the realm_ssid ID");
+                
+                #Get the realm_ssid id
+                my $ssid_id = 0;
+                $stmt_ssid_id->execute($ssid,$realm_id);        
+                while(my $row = $stmt_ssid_id->fetchrow_hashref()){        
+                    $ssid_id = $row->{'id'};
                 }
-                $stmt_pmk_list->finish();
-                if($match_found == 0){
-                    $RAD_REPLY{'Reply-Message'} = "No PPSK Match Found";
-                    $return = RLM_MODULE_REJECT;               
-                }            
-            }                     
+                $stmt_ssid_id->finish();
+                
+                if($ssid_id){
+                    &radiusd::radlog("2", "Found Realm ID $realm_id and ssid_id $ssid_id. We can try to get the LIST OF PPSKs");
+                    $stmt_pmk_list->execute($ssid_id,$realm_id); 
+                    while(my $row = $stmt_pmk_list->fetchrow_hashref()){
+                        if(process_row($EAPOL1,$EAPOL2,$ssid,$row->{'ppsk'},$row)){
+                            #Formulate the reply
+                            formulate_reply($row,$realm_id);
+                            $match_found = 1;
+                            last OUTERLOOP;
+                        }
+                    }
+                    $stmt_pmk_list->finish();
+                }                     
+            }
+            if($match_found == 0){
+                $RAD_REPLY{'Reply-Message'} = "No PPSK Match Found";
+                $return = RLM_MODULE_REJECT;               
+            }            
         }        
     }
 }
